src/Paket.Core/Installation/InstallModel.fs

```diff
@@ -0,0 +1,813 @@
+    member this.GetRuntimeAssemblies graph rid target = InstallModel.getRuntimeAssemblies graph rid target this
+    member this.GetRuntimeLibraries graph rid target = InstallModel.getRuntimeLibraries graph rid target this
+
+    [<Obsolete("usually this should not be used, use GetLegacyReferences for the full .net and GetCompileReferences for dotnetcore")>]
+    member this.GetLibReferences frameworkIdentifier = InstallModel.getLegacyPlatformReferences frameworkIdentifier this
+
+    member this.GetTargetsFiles target = InstallModel.getTargetsFiles target this
+
+    member this.GetAllLegacyFrameworkReferences () = InstallModel.getAllLegacyFrameworkReferences this
+    member this.GetAllLegacyReferences () = InstallModel.getAllLegacyReferences this
+    member this.GetAllLegacyReferenceAndFrameworkReferenceNames () =
+        this.GetAllLegacyFrameworkReferences() |> Seq.map (fun r -> r.Name)
+        |> Seq.append (this.GetAllLegacyReferences() |> Seq.map (fun r -> r.Name))
+        |> Set.ofSeq
+
+    [<Obsolete("usually this should not be used, use CalcLegacyReferencesFolders for the full .net and CalcReferencesFolders for dotnetcore")>]
+    member this.CalcLibFolders libs = InstallModel.calcLegacyReferenceLibFolders libs
+    member this.CalcLegacyReferencesFolders libs = InstallModel.calcLegacyReferenceLibFolders libs
+    member this.CalcReferencesFolders libs = InstallModel.calcReferenceFolders libs
+
+
+    member this.AddLibReferences (libs, references) = InstallModel.addLibReferences libs references this
+
+    member this.AddReferences libs = InstallModel.addLibReferences libs NuspecReferences.All this
+
+    member this.AddAnalyzerFiles analyzerFiles = InstallModel.addAnalyzerFiles analyzerFiles this
+
+    member this.AddTargetsFile(path, file) = InstallModel.addTargetsFile path file this
+
+    member this.AddTargetsFiles targetsFiles = InstallModel.addTargetsFiles targetsFiles this
+
+    member this.AddFrameworkAssemblyReference reference = InstallModel.addFrameworkAssemblyReference this reference
+
+    member this.AddFrameworkAssemblyReferences references = InstallModel.addFrameworkAssemblyReferences references this
+
+    member this.FilterBlackList () = InstallModel.filterBlackList this
+
+    member this.FilterExcludes excludes = InstallModel.filterExcludes excludes this
+
+    member this.FilterReferences(references) = InstallModel.filterReferences references this
+
+    member this.ApplyFrameworkRestrictions restrictions = InstallModel.applyFrameworkRestrictions restrictions this
+
+    member this.RemoveIfCompletelyEmpty() = InstallModel.removeIfCompletelyEmpty this
+
+    static member CreateFromLibs(packageName, packageVersion, frameworkRestrictions:FrameworkRestriction list, libs : UnparsedPackageFile seq, targetsFiles, analyzerFiles, nuspec : Nuspec) =
+        InstallModel.createFromLibs packageName packageVersion frameworkRestrictions libs targetsFiles analyzerFiles nuspec
```

Comment:
Wow not even git recognizes that this file existed before and thinks it was deleted and added again :)